includes/Gateway/Cash_App_Pay_Gateway.php

```diff
@@ -0,0 +1,1226 @@
+		</form>
+		<?php
+	}
+
+	/**
+	 * Return the gateway-specifics JS script handle.
+	 *
+	 * @since x.x.x
+	 * @return string
+	 */
+	protected function get_gateway_js_handle() {
+		return 'wc-' . $this->get_id_dasherized();
+	}
+
+	/**
+	 * Enqueue the gateway-specific assets if present, including JS, CSS, and
+	 * localized script params
+	 *
+	 * @since x.x.x
+	 */
+	protected function enqueue_gateway_assets() {
+		$is_checkout = is_checkout() || ( function_exists( 'has_block' ) && has_block( 'woocommerce/checkout' ) );
+
+		// bail if not a checkout page or cash app pay is not enabled
+		if ( ! ( $is_checkout || $this->is_configured() ) ) {
+			return;
+		}
+
+		if ( $this->get_plugin()->get_settings_handler()->is_sandbox() ) {
+			$url = 'https://sandbox.web.squarecdn.com/v1/square.js';
+		} else {
+			$url = 'https://web.squarecdn.com/v1/square.js';
+		}
+
+		wp_enqueue_script( 'wc-' . $this->get_plugin()->get_id_dasherized() . '-payment-form', $url, array(), Plugin::VERSION, true );
+
+		parent::enqueue_gateway_assets();
+
+		// Render Payment JS
+		$this->render_js();
+	}
+
+	/**
+	 * Validates the entered payment fields.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return bool
+	 */
+	public function validate_fields() {
+		$is_valid = true;
+
+		try {
+			if ( ! Square_Helper::get_post( 'wc-' . $this->get_id_dasherized() . '-payment-nonce' ) ) {
+				throw new \Exception( 'Payment nonce is missing.' );
+			}
+		} catch ( \Exception $exception ) {
+
+			$is_valid = false;
+
+			Square_Helper::wc_add_notice( __( 'An error occurred, please try again or try an alternate form of payment.', 'woocommerce-square' ), 'error' );
+
+			$this->add_debug_message( $exception->getMessage(), 'error' );
+		}
+
+		return $is_valid;
+	}
+
+	/** Admin methods *************************************************************************************************/
+	/**
+	 * Adds admin notices.
+	 *
+	 * @since x.x.x
+	 */
+	public function add_admin_notices() {
+		$base_location      = wc_get_base_location();
+		$is_plugin_settings = $this->get_plugin()->is_payment_gateway_configuration_page( $this->get_id() );
+		$is_connected       = $this->get_plugin()->get_settings_handler()->is_connected() && $this->get_plugin()->get_settings_handler()->get_location_id();
+		$is_enabled         = $this->is_enabled() && $is_connected;
+
+		// Add a notice for cash app pay if the base location is not the US.
+		if ( ( $is_enabled || $is_plugin_settings ) && isset( $base_location['country'] ) && 'US' !== $base_location['country'] ) {
+
+			$this->get_plugin()->get_admin_notice_handler()->add_admin_notice(
+				sprintf(
+					/* translators: Placeholders: %1$s - <strong> tag, %2$s - </strong> tag, %3$s - 2-character country code, %4$s - comma separated list of 2-character country codes */
+					__( '%1$sCash App Pay (Square):%2$s Your base country is %3$s, but Cash App Pay (Square) can’t accept transactions from merchants outside of US.', 'woocommerce-square' ),
```

Comment:
We can probably remove `(Square)` here as most merchants/shoppers recognize Cash App as the brand and not that its owned by Square

Reply:
Actually, let's just remove the second reference of `(Square)` as the first does make it clear for the merchant that its coming from the Square gateway (in case they have multiple gateways active)

Reply:
Nice catch! handled in https://github.com/woocommerce/woocommerce-square/pull/53/commits/bab11c3d255371017fa93d9651544ecd5494516a